modules/cms-kit/src/Volo.CmsKit.Public.Application/Volo/CmsKit/Public/Ratings/RatingPublicAppService.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Volo.Abp.Application.Dtos;
+using Volo.Abp.Application.Services;
+using Volo.Abp.Authorization;
+using Volo.Abp.Users;
+using Volo.CmsKit.Ratings;
+using Volo.CmsKit.Users;
+
+namespace Volo.CmsKit.Public.Ratings
+{
+    public class RatingPublicAppService : ApplicationService, IRatingPublicAppService
+    {
+        protected IRatingRepository RatingRepository { get; }
+        public ICmsUserLookupService CmsUserLookupService { get; }
+
+        public RatingPublicAppService(IRatingRepository ratingRepository, ICmsUserLookupService cmsUserLookupService)
+        {
+            RatingRepository = ratingRepository;
+            CmsUserLookupService = cmsUserLookupService;
+        }
+
+        public virtual async Task<ListResultDto<RatingDto>> GetListAsync(string entityType, string entityId)
+        {
+            var ratings = await RatingRepository.GetListAsync(entityType, entityId);
+            var ratingDto = ObjectMapper.Map<List<Rating>, List<RatingDto>>(ratings);
+
+            return new ListResultDto<RatingDto>(ratingDto);
+        }
+
+        [Authorize]
+        public virtual async Task<RatingDto> CreateAsync(string entityType, string entityId, CreateRatingInput input)

```

Comment:
A user should not be able to create multiple stars for the same entity.
I suggest to delete the UpdateAsync method and handle the update in this (CreateAsync) method (I mean you update or create star for an entity).